### PR TITLE
Fixing #55 and #81

### LINKED ISSUE #55: Replace all tidyr::gather() code in help files with pivot_longer()
Some of the help/man files have `roxygen2` examples that convert data as originally given in [538's GitHub](https://github.com/fivethirtyeight/data) to "tidy" format using `tidyr::gather()`. However, `tidyr` now includes much more intuitively-named and easier to use functions `pivot_longer()` and `pivot_wider()`. See this tidyverse.org [blog post](https://tidyr.tidyverse.org/dev/articles/pivot.html) for more information.

In the case of `fivethirtyeight` package, we should replace all `gather()` code with `pivot_longer()`. For example, if you run `?drinks` you'll currently see in the examples:
```{r}
library(dplyr)
library(tidyr)
library(stringr)
drinks_tidy <- drinks %>%
  gather(type, servings, -c(country, total_litres_of_pure_alcohol)) %>%
  mutate(
    type = str_sub(type, start=1, end=-10)
  ) %>%
  arrange(country, type)
```
This needs to be updated with:
```{r}
library(fivethirtyeight)
library(dplyr)
library(tidyr)
library(stringr)
drinks_tidy <- drinks %>%
  pivot_longer(cols = ends_with("servings"), names_to = "type", values_to = "servings") %>%
  mutate(type = str_sub(type, start=1, end=-10)) %>%
  arrange(country, type)
```
A search of `gather` in all `R/data_X.R` files will locate the `roxygen2` code for all such cases. For example, the `roxygen2` code for `drinks` is in [`data_albert.R`](https://github.com/rudeboybert/fivethirtyeight/blob/42d2bbb21e4dce3492d5515f0934c7a435bfd831/R/data_albert.R#L129).

@beanumber Let me know if you have someone who can do this, and I'll explicitly make them an assignee.

### LINKED ISSUE #81: Use \dontrun{} in @examples roxygen code instead of if(FALSE)
For example, in `house_district_forecast` help file in [`R/data_starry.R`](https://github.com/rudeboybert/fivethirtyeight/blob/master/R/data_starry.R#L157), we use `if(FALSE)` to not have code in example run. Change all these to `\dontrun{}`

### PR DESCRIPTION
**Pull request by rudeboybert**
Hey @mariumtapal, I'm going to create the PR so that I can easily see side-by-side diffs of all changed files, but I won't review the code for merging until you explicitly tag me for a review (See right-hand menubar)

### CODE REVIEW

**Comment by mariumtapal** [comment]
@rudeboybert I can't tag you for a review since you created this PR!

File: R/data_maggie.R [hunk 1]
```diff
@@ -168,9 +169,9 @@
 #' library(dplyr)
 #' library(tidyr)
 #' nfl_fandom_surveymonkey_tidy <- nfl_fandom_surveymonkey %>%
-#'   gather(key = race_party, value = percent,
-#'          -c("team", "total_respondents", "gop_percent", "dem_percent",
-#'             "ind_percent", "white_percent", "nonwhite_percent")) %>%
+#'   pivot_longer(-c("team", "total_respondents", "gop_percent", "dem_percent",
```
Comment (rudeboybert) [inline R/data_maggie.R:172]: I tried running a local package check of your branch, and I’m getting errors. In particular relating to the `pivot_longer()` code like the one here. Could you fix these? Also, it’s generally good practice to run one final local check before requesting a review

**Comment by rudeboybert** [comment]
Also another minor comment @mariumtapal. For all intents and purposes, your above work gets the job done. but whenever possible, it's better to break down your commits into smaller and more modular pieces. For ex in this case one commit for #55 and a separate one for #81. Now that I think about it, you could even argue it might be better to create two separate branches/PRs. That way if you want to revert only one of the two changes, you can do so easily. Let's leave as is this go round

File: R/data_maggie.R (hunk 1 above)
Comment (mariumtapal) [reply R/data_maggie.R:172]: fixed!

File: DESCRIPTION [hunk 2]
```diff
@@ -42,7 +42,7 @@ Encoding: UTF-8
 LazyData: true
 URL: https://github.com/rudeboybert/fivethirtyeight
 BugReports: https://github.com/rudeboybert/fivethirtyeight/issues
-RoxygenNote: 7.0.2
+RoxygenNote: 7.1.0
 Suggests:
     ggplot2,
     dplyr,
```
Comment (ismayc) [inline DESCRIPTION:48]: @rudeboybert In `Suggests:` we should probably update `tidyr` to be `tidyr (>= 1.0.0)` to reflect the switch to `pivot_*`, right?

**Review by ismayc — CHANGES_REQUESTED** [review]
Everything looks good. I left a comment about the DESCRIPTION file.

File: DESCRIPTION (hunk 2 above)
Comment (rudeboybert) [reply DESCRIPTION:48]: good point. will fix

**Review by ismayc — APPROVED** [review]

**Pull request merged**